magicctapipe/scripts/mars/tests/compare_hillas_stereo_params.py

```diff
@@ -0,0 +1,291 @@
+import pandas as pd
+import uproot
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import numpy as np
+import yaml
+# from astropy.coordinates import angular_separation, EarthLocation, AltAz, SkyCoord
+# from astropy.time import Time
+# from astropy import units as u
+# from math import *
+from labels_and_units_for_params_comparison import (
+    mars_params,
+    mcp_params,
+    scale_factors,
+    threshold,
+    labels_and_units,
+    scaling,
+)
+from pathlib import Path
+
+
+def compare_hillas_stereo_parameters(
+    config_file="config.yaml", params_key="events/parameters", plot_image=False
+):
+    """
+    This fuction compares the values of the hillas and stereo parameters for MARS and magic-ctapipe. It returns a list of True/False
+    corresponding to the input compared parameters.
+    True : Error found, False : No Errors
+    The second parameter returned is the list of fractions of events with a problem in a given parameter
+    ----------
+    config_file: path of config file
+    params_key: key in the dl1_stereo files (magic-cta-pipe output) where the hillas and stereo params are saved
+    plot_image: if True, a plot is created for each parameter and saved
+    """
+
+    config = yaml.safe_load(open(config_file, "r"))
+    filename_no_ext = Path(config["MARS-input"]["MARS-path"]).stem
+
+    # --------------
+    # read mcp data
+    # --------------
+    mcp_file = config["magic-cta-pipe-input"]["MCP-path"]
+
+    df_mcp = pd.read_hdf(mcp_file, key=params_key)
+
+    # filter for subrun
+    # df_mcp = df_mcp.loc[df_mcp["obs_id"] == subrun]
+
+    print(f"Number of events before cuts {df_mcp.shape[0]}")
+
+    # LON_ORM = u.Quantity(-17.89064, u.deg)
```

Comment:
there is a lot of commented code here, will it be used for the further developments?

Reply:
This part of the code is used to apply a theta2 cut when comparing the impact. But when applied only few events survived. Maybe it is better to just not use it at all?

Reply:
for comparisons on the data such a cut would indeed remove nearly all the events so it does not make much sense. one could keep it for the MCs, but only if we really some a problem, i.e. impact matches nicely for closeby events, but ends up with much larger spread between both methods for high theta events (and actually we also have two 'theta' values - from MARS and from MCP that could be different as well)